src/instantiator.js

```diff
@@ -105,6 +105,13 @@ function instantiate(schema, options) {
       return;
     }
     var i;
+    // Check if type is array of types
+    // if so, set type to first in list of types
+    if (Array.isArray(obj.type)) {
```

Comment:
Please enter this snippet to a separate function, so instead of lines 108-115, we'll have:

```javascript
var type = getObjectType(obj);
```